Issues/56 Test facts based on privilege of auth

kdelee opened the pull request:
@elyezer 

This "works" but has a couple rough points:

One is the fact that a lot of the jboss facts will come back '(...factname... not found)' if JBoss is in a location that the scan user cannot read. This is the case on most of our machines, but you would actually get the same data if the files were in directories that the scan user could read (like its home directory).

Also, I'm not sure if the underprivileged user will get the same count on jboss processes, it seems not but I'm not 100% sure this is a privilege issue, and I don't feel great about doing some kind of arithmetic on how many fewer jboss processes the user will find.


All in all, I think our test results will now provide a lot more insight as to how many facts we are having issues with when we see this:
```
test_scan.py::test_facts[IPADDR-PROFILENAME-['scanbasic']-virt.num_guests] PASSED
test_scan.py::test_facts[IPADDR-PROFILENAME-['scanbasic']-virt.num_running_guests] PASSED
test_scan.py::test_facts[IPADDR-PROFILENAME-['scanbasic']-virt.type] PASSED
test_scan.py::test_facts[IPADDR-PROFILENAME-['scanbasic']-virt.virt] PASSED

2 failed, 816 passed in 365.77 seconds
```

That is just with a small sample of the test machines. It is upwards of 3000 "test" items when I scan all vcenter machines.

Let me know what you think!

elyezer commented on camayoc/tests/remote/rho/conftest.py:
```diff
@@ -0,0 +1,149 @@
+    cfg = config.get_config()
+    facts_to_test = []
+    for profile in profiles_to_scan:
+        cwd = os.getcwd()
```
It seems that you are duplicating what the isolated filesystem does here, so I think it would be better to use the context manager here https://github.com/quipucords/camayoc/blob/master/camayoc/utils.py#L20

kdelee replied:
Thanks for bringing this up, when I realized I could not use the fixture did not connnect the dots that I could just use the context manager.

elyezer commented on camayoc/tests/remote/rho/conftest.py:
```diff
@@ -0,0 +1,149 @@
+            assert rho_scan.exitstatus == 0, logfile
+            assert os.path.isfile(reportfile)
+        except (AssertionError, pexpect.exceptions.EOF):
+            # The scan failed for some reason.
```
I think this is good for now but we may want to log the issues to help finding some errors later.

elyezer commented on camayoc/tests/remote/rho/test_scan.py:
```diff
@@ -90,42 +86,24 @@ def test_scan(isolated_filesystem, profile):
+    :id: d9eb29bd-1b61-421a-b680-f494e868b11e
+    :description: Test the facts from a scan report
+    :steps:
+            1) Test facts collected from scans against expected data
```
I usually indent the line using 4 spaces, please check if this is being rendered on the docs as a numbered list. It should but I think it worth double checking.

kdelee commented on camayoc/tests/remote/rho/conftest.py:
```diff
@@ -0,0 +1,149 @@
+
+import csv
+import os
+import pexpect
```
Yes, thanks for reminding me, I'll do this.

kdelee commented:
@elyezer I have updated with changes addressing your comments. Thanks for the feedback. 

I am updating the spreadsheets on the shared drive for generating the config files and am going to test one more time. 

I am thinking that in the next sprint when we tackle "batching" our automation so we are not testing all the machines at once (having all the VMs on at the same time), I may want to also address breaking out different jobs for the different privilege levels. 

Really its just a matter of organization and making our results readable/informative.

kdelee commented:
@elyezer  Thanks! I'll merge and update the config files on jenkins and we'll see how it goes.

Talking to Og has also given me some ideas about future improvements, but I want to see this in action for a little while.
